pkg/openapi/crdSync.go

```diff
@@ -119,13 +125,14 @@ func (c *crdSync) sync() {
 }
 
 func (c *crdSync) updateInClusterKindToAPIVersions() error {
-	_, apiResourceLists, err := c.client.Discovery().DiscoveryCache().ServerGroupsAndResources()
-	if err != nil {
+	util.OverrideRuntimeErrorHandler()
+	_, apiResourceLists, err := discovery.ServerGroupsAndResources(c.client.Discovery().DiscoveryInterface())
+
+	if err != nil && !strings.Contains(err.Error(), skipErrorMsg) {
 		return errors.Wrapf(err, "fetching API server groups and resources")
 	}
-
-	preferredAPIResourcesLists, err := c.client.Discovery().DiscoveryCache().ServerPreferredResources()
-	if err != nil {
+	preferredAPIResourcesLists, err := discovery.ServerPreferredResources(c.client.Discovery().DiscoveryInterface())
+	if err != nil && !strings.Contains(err.Error(), skipErrorMsg) {
```

Comment:
do we still need this?

Reply:
Yes, we need that. we are getting error, only we are suppressing those.
also I have tested if we remove will get error in logs. so we required those check.
```sh
E0701 02:29:18.601582       1 crdSync.go:113]  "msg"="sync failed, unable to update in-cluster api versions" "error"="fetching API server groups and resources: unable to retrieve the complete list of server APIs: external.metrics.k8s.io/v1beta1: Got empty response for: external.metrics.k8s.io/v1beta1"  
E0701 02:29:33.840345       1 crdSync.go:113]  "msg"="sync failed, unable to update in-cluster api versions" "error"="fetching API server groups and resources: unable to retrieve the complete list of server APIs: external.metrics.k8s.io/v1beta1: Got empty response for: external.metrics.k8s.io/v1beta1"  
E0701 02:29:49.131088       1 crdSync.go:113]  "msg"="sync failed, unable to update in-cluster api versions" "error"="fetching API server groups and resources: unable to retrieve the complete list of server APIs: external.metrics.k8s.io/v1beta1: Got empty response for: external.metrics.k8s.io/v1beta1"  
E0701 02:30:04.377697       1 crdSync.go:113]  "msg"="sync failed, unable to update in-cluster api versions" "error"="fetching API server groups and resources: unable to retrieve the complete list of server APIs: external.metrics.k8s.io/v1beta1: Got empty response for: external.metrics.k8s.io/v1beta1" 
```